File: aula4/Calculadora.py
class Calculadora:

    # Método para somar dois ou mais números
    def somar(self, *args):
        return sum(args)

    # Método para somar todos os números de uma lista de inteiros
    def somar_lista(self, numeros: list) -> int:
        return sum(numeros)

    # Método para multiplicar dois ou mais números
    def multiplicar(self, *args):
        produto = 1
        for num in args:
            produto *= num
        return produto

    # Método para multiplicar todos os números de uma lista de inteiros
    def multiplicar_lista(self, numeros: list) -> int:
        produto = 1
        for num in numeros:
            produto *= num
        return produto


# Testando os métodos da Calculadora
if __name__ == "__main__":
    calc = Calculadora()
    
    # Exemplo de soma com dois inteiros
    print("Soma de 2 e 3:", calc.somar(2, 3))
    
    # Exemplo de soma com três inteiros
    print("Soma de 2, 3 e 4:", calc.somar(2, 3, 4))
    
    # Exemplo de soma com números do tipo float
    print("Soma de 2.5 e 3.5:", calc.somar(2.5, 3.5))
    
    # Exemplo de soma com uma lista de inteiros
    numeros = [1, 2, 3, 4, 5, 6, 7, 8, 9]
    print("Soma da lista:", calc.somar_lista(numeros))
    
    # Exemplo de multiplicação com dois inteiros
    print("Multiplicação de 2 e 3:", calc.multiplicar(2, 3))
    
    # Exemplo de multiplicação com três inteiros
    print("Multiplicação de 2, 3 e 4:", calc.multiplicar(2, 3, 4))
    
    # Exemplo de multiplicação com números do tipo float
    print("Multiplicação de 2.5 e 3.5:", calc.multiplicar(2.5, 3.5))
    
    # Exemplo de multiplicação com uma lista de inteiros
    print("Multiplicação da lista:", calc.multiplicar_lista(numeros))

File: aula4/Calculadora.cpp
#include <iostream>
#include <vector>
using namespace std;

class Calculadora{

public:

    int somar(int a, int b){
        return a + b;
    }

    int somar(int a, int b, int c){
        return a + b + c;
    }

    double somar(double a, double b){
        return a + b;
    }

    int somar(vector<int>& nums){
        int soma = 0;
        for(int num : nums)
            soma += num;
        return soma;
    }

    double somar(vector<double>& nums){
        double soma = 0.0;
        int tam = nums.size();
        for(int i = 0; i < tam; i ++)
            soma += nums[i];
        return soma;
    }

    int multiplicar(int a, int b){
        return a * b;
    }

    int multiplicar(int a, int b, int c){
        return a * b * c;
    }

    int multiplicar(vector<int>& nums){
        int soma = 1;
        for(int num : nums)
            soma *= num;
        return soma;
    }

};

int main() {
    Calculadora calc;

    cout << "Soma de 2 e 3: " << calc.somar(2, 3) << endl;
    cout << "Soma de 2, 3 e 4: " << calc.somar(2, 3, 4) << endl;
    cout << "Soma de 2.5 e 3.5: " << calc.somar(2.5, 3.5) << endl;

    vector<int> numeros = {1, 2, 3, 4, 5, 6, 7, 8, 9};
    cout << "Soma do vetor: " << calc.somar(numeros) << endl;

    cout << "Multiplicação de 2 e 3: " << calc.multiplicar(2, 3) << endl;
    cout << "Multiplicação de 2, 3 e 4: " << calc.multiplicar(2, 3, 4) << endl;
    cout << "Multiplicação de 2.5 e 3.5: " << calc.multiplicar(2.5, 3.5) << endl;
    cout << "Multiplicação do vetor: " << calc.multiplicar(numeros) << endl;

    return 0;
}